### PR TITLE
python-and-data-science-tools-HYF-homework

### PR DESCRIPTION
**Pull request by Mekdes1**

### CODE REVIEW

File: week1/main.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+age = 25
+print(f'Name: {name} Age: {age}')
+
+# Create a list of your 5 favorite movies and store it in the variable
+my_favorite_movies=['Ratatouille','Manchester by the Sea','The Godfather','Rear Window','Boyhood',]
+print(f'My favorite movies are: {my_favorite_movies}')
+
+#Write a Python program to display the first and last colors from the following list.
+color_list = ["Red","Green","White" ,"Black"]
+first_and_last_colors = (f'{color_list[0]} and { color_list[-1]}')
```
Comment (Rassibassi) [inline week1/main.py]: the brackets `(` `)` are not necessary

File: week1/main.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+sample_dictionary[2] = 30
+print(sample_dictionary)
+
+#Write a Python program to calculate body mass index.
+def body_mass_index_calculator():
+    superscript = str.maketrans("0123456789", "⁰¹²³⁴⁵⁶⁷⁸⁹")
+    weight = float(input('Enter your weight (kg): '))
+    height = float(input(f'Enter your height ({"m2".translate(superscript)}): '))
+    body_mass_index = weight / height ** 2
+    return  (f'your body mass index is: {round(body_mass_index, 2)}')
```
Comment (Rassibassi) [inline week1/main.py]: the brackets `(` `)` are not necessary

File: week1/main.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+guess_number_game()
+
+#Create a tuple with different data types
+sample_tuple = ('tuple', 50, 1.4, True)
+print(sample_tuple)
+
+#Create a list of 5 city names and convert it into tuples.
+def convert_to_tuples(list):
+    return tuple(list)
+sample_list = ['Dubai','Paris',8,9,10]
```
Comment (Rassibassi) [inline week1/main.py:47]: `convert_to_tuples` only wraps `tuple` which is a function itself. my not using `tuple` directly

File: week1/main.py [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+#Accept a string from user and remove the characters which have odd index values of a given string and print them
+
+def odd_index_remover():
+    given_string= str(input('Please enter your string: '))
+    string_with_removed_index= ''
+
+    for i in range(len(given_string)):
+        if(i % 2 == 0):
+           string_with_removed_index= (f'{string_with_removed_index}{given_string[i]}')
```
Comment (Rassibassi) [inline week1/main.py]: This line can be done like:
```python
string_with_removed_index += given_string[i]
```

File: week1/main.py [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+    print(f'Given string:- {given_string}')
+    print(f'Sting with removed indes:- {string_with_removed_index}')
+
+
+odd_index_remover()
```
Comment (Rassibassi) [inline week1/main.py:74]: Python can do this more elegant or for a _Pythonista_, there is a more _pythonic_ way of doing this.

In Python **a string is basically a list of characters**, or in other words, in Python like a list also a string is **iterable**,
and a list can be looped through like this:
```python
sample_list = [100,200,300]
for sample_int in sample_list:
    print(sample_int)
```
Do you see that there is no indexing `sample_list[i]` necessary anymore? You can do the same with a string

Now, you might have also seen that we lost the index number `i` within the loop. Yet, there is also a pythonic way of getting a running number into your for loop, see here:
```python
sample_list = [100,200,300]
for i, sample_int in enumerate(sample_list):
    print(i, sample_int)
```

Can you use my feedback and improve your code?

File: week1/main.py [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+print(convert_to_non_duplicated_list(sample_duplicated_list))
+
+#Accept a string from user and remove the characters which have odd index values of a given string and print them
+
+def odd_index_remover():
+    given_string= str(input('Please enter your string: '))
+    string_with_removed_index= ''
+
+    for i in range(len(given_string)):
+        if(i % 2 == 0):
```
Comment (Rassibassi) [inline week1/main.py]: brackets not needed :)

**Review by Rassibassi — COMMENTED** [review]
Good job, see inline comments

File: week1/main.py (hunk 5 above)
Comment (Mekdes1) [reply week1/main.py:74]: Yes , thank you for your feedback.

**Pull request merged**